dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/javadsl/DynamoDb.scala

```diff
@@ -32,32 +32,21 @@ object DynamoDb {
     scaladsl.DynamoDb.flow(parallelism)(client, operation).asJava
 
   /**
-   * Create a Flow that emits a response for every request to DynamoDB.
-   * A successful response is wrapped in [scala.util.Success] and a failed
-   * response is wrapped in [scala.util.Failure].
+   * Create a `FlowWithContext` that emits a response for every request to DynamoDB.
+   * A successful response is wrapped in [[scala.util.Success]] and a failed
+   * response is wrapped in [[scala.util.Failure]].
    *
-   * The returned flow is meant to compose easily with an Akka Stream RetryFlow
-   * which can retry requests, that have responses wrapped in [scala.util.Try].
+   * The context is merely passed through to the emitted element.
    *
    * @param parallelism maximum number of in-flight requests at any given time
-   *
-   * @tparam State the type of the pass-through value that is taken together with
-   *               requests and is emitted together with responses. Can be used,
-   *               for example:
-   *                 * to correlate a request with a response
-   *                 * to track number of retries
-   *                 * to store request to be issued in the case of retry
+   * @tparam Ctx context (or pass-through)
    */
-  def tryFlow[In <: DynamoDbRequest, Out <: DynamoDbResponse, State](
+  def flowWithContext[In <: DynamoDbRequest, Out <: DynamoDbResponse, Ctx](
       client: DynamoDbAsyncClient,
       operation: DynamoDbOp[In, Out],
       parallelism: Int
-  ): Flow[akka.japi.Pair[In, State], akka.japi.Pair[Try[Out], State], NotUsed] =
-    Flow
-      .create[akka.japi.Pair[In, State]]()
-      .map(func(p => (p.first, p.second)))
-      .via(scaladsl.DynamoDb.tryFlow(parallelism)(client, operation))
-      .map(func(t => akka.japi.Pair.create(t._1, t._2)))
+  ): FlowWithContext[In, Ctx, Try[Out], Ctx, NotUsed] =
```

Comment:
I saw your note about using `Try` in the Java DSL. How come we even expose a `Try[Out]`? We could leave it up to the user to recover the stream.  Another option is to use `Either`, and then add a simple Alpakka implementation (and propose it upstream in Akka's `japi` package).  The only other Java alternatives I can think of are `Optional`, which loses information, or `japi.Pair` which seems like the wrong fit.

Reply:
It is this new API that exposes the `Try`, so we might remove it before releasing. Offering a Java counterpart might be a solution.

Reply:
The old API still had an output type of `akka.japi.Pair[Try[Out], State]`, which makes sense I guess since it was called the `tryFlow`.

Reply:
Ah, that "old API" was recently added by @2m, it wasn't part of an Alpakka release.

Reply:
Oh, I see.  Ok.